lib/child_process.js

```diff
@@ -176,17 +174,26 @@ exports.execFile = function(file /*, args, options, callback*/) {
 
     if (!callback) return;
 
-    // merge chunks
-    var stdout;
-    var stderr;
-    if (!encoding) {
-      stdout = Buffer.concat(_stdout);
-      stderr = Buffer.concat(_stderr);
-    } else {
-      stdout = _stdout;
-      stderr = _stderr;
+    var stdout = Buffer.concat(_stdout, stdoutLen);
+    var stderr = Buffer.concat(_stderr, stderrLen);
+
+    var stdoutEncoding = encoding;
+    var stderrEncoding = encoding;
+
+    if (stdoutState.decoder) {
```

Comment:
Maybe I spoke too soon... it may be possible for `stdoutState` to be `undefined` if `child.stdout` for whatever reason (there are checks for it throughout the code in here), so I guess we will need to prefix with `stdoutState && `. Same for stderr.